scipy/stats/tests/test_distributions.py

```diff
@@ -3067,64 +3100,121 @@ def test_pdf_nolan_samples(self):
                 1,0 # gamma, delta
                 2 # output file
         """
-        fn = os.path.abspath(os.path.join(os.path.dirname(__file__),
-                                          'data/stable-pdf-sample-data.npy'))
-        data = np.load(fn)
+        data = np.load(
+            Path(__file__).parent /
+            'data/levy_stable/stable-Z1-pdf-sample-data.npy'
+        )
 
-        data = np.core.records.fromarrays(data.T, names='x,p,alpha,beta')
+        data = np.core.records.fromarrays(data.T, names='x,p,alpha,beta,pct')
 
         # support numpy 1.8.2 for travis
         npisin = np.isin if hasattr(np, "isin") else np.in1d
 
+        # some tests break on linux 32 bit
+        uname = platform.uname()
+        is_linux_32 = uname.system == 'Linux' and uname.machine == 'i686'
+        platform_desc = "/".join(
+            [uname.system, uname.machine, uname.processor])
```

Comment:
A better way to handle test cases that break in some situations is to use pytest's [xfail](https://docs.pytest.org/en/6.2.x/skipping.html#xfail-mark-test-functions-as-expected-to-fail), with a condition set. 

The example in the pytest documentation is
```python
@pytest.mark.xfail(sys.platform == "win32", reason="bug in a 3rd party library")
def test_function():
```

see this [stackoverflow answer](https://stackoverflow.com/a/30575822/992102) for how to do this for individual test cases instead of an entire test.